core/src/test/scala/org/apache/spark/util/collection/FixedHashObject.scala

```diff
@@ -22,4 +22,8 @@ package org.apache.spark.util.collection
  */
 case class FixedHashObject(v: Int, h: Int) extends Serializable {
   override def hashCode(): Int = h
+  override def equals(other: Any): Boolean = other match {
```

Comment:
Aha you're right about that. Right, this actually represents no behavior change.